web/themes/custom/novel/novel.theme

```diff
@@ -86,3 +86,71 @@ function novel_preprocess_dpl_react_app__search_result(array &$variables): void
   // That enables us to vary the "searching" title line.
   $variables['#cache']['contexts'][] = "url.query_args:q";
 }
+
+/**
+ * Implements hook_preprocess_HOOK() for node templates.
+ *
+ * Preprocesses variables for node templates.
+ *
+ * Includes generating ticket pricing display in 'event_list_teaser' view mode.
+ *
+ * @param mixed[] $variables
+ *   The variables array for the node template.
+ *
+ * @throws \Safe\Exceptions\ArrayException
+ */
+function novel_preprocess_node(array &$variables):void {
+  if ($variables['node']->hasField('field_ticket_categories')) {
+    $ticket_categories = $variables['node']->get('field_ticket_categories')->referencedEntities();
+    $prices = [];
+
+    // Collect prices from ticket categories.
+    foreach ($ticket_categories as $category) {
+      if ($category->hasField('field_ticket_category_price') && !$category->get('field_ticket_category_price')->isEmpty()) {
+        $price = $category->get('field_ticket_category_price')->value;
+        $prices[] = (floatval($price) == intval($price)) ? intval($price) : floatval($price);
+      }
+    }
+
+    $variables['ticket_price_display'] = novel_get_ticket_price_display($prices);
```

Comment:
Please reconsider all this price handling logic in the theme.

We already have [a class for handling price formatting](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/b47d1118ca6c79a5c062af5c41ba5d08d321cc4f/web/modules/custom/dpl_event/src/PriceFormatter.php#L14-L14). It handles formatting single prices. I think we should extend it to handle an array of prices. This provides a number of benefits:

  - The class uses `BigDecimal` to handle juggling of floats and ints for you.
  - You have a translator available so you can translate strings. The current ones in the code are hardcoded to Danish.
  - We can implement a unit test of the code by extending the current test. This makes it a lot easier to verify that the code it working as intended
  - This moves the code away from the template. I have previously heard @rasben request that we move logic to modules instead of the `.theme` file where possible.

Reply:
Yeah, I would argue this is complicated enough that it should be in a module, rather than `.theme`

Reply:
I have moved this logic from the theme file, to the dpl_event module. 

In terms of price calculation, i've added methods and refactored priceFormatter.php 

I have also added tests for the new methods. 

**Important**

I got an error with phpcs specified in the commit that i was not able to fix. 
Please remove the line specified in the commit message for the tests and try to replicate.